src/android/Fingerprint.java

```diff
@@ -71,212 +67,241 @@ public int getValue() {
+                return new Error(
+                        PluginError.BIOMETRIC_HARDWARE_NOT_SUPPORTED.getValue(),
+                        PluginError.BIOMETRIC_HARDWARE_NOT_SUPPORTED.name());
+            case BiometricManager.BIOMETRIC_ERROR_NONE_ENROLLED:
+                return new Error(
+                        PluginError.BIOMETRIC_FINGERPRINT_NOT_ENROLLED.getValue(),
+                        PluginError.BIOMETRIC_FINGERPRINT_NOT_ENROLLED.name());
+            default:
+                return null;
         }
     }
 
-    @Override
-    public void onAuthenticationSuccessful() {
-        Log.e(TAG, "biometric_success");
-        //mCallbackContext.success("biometric_success");
-        sendSuccess("biometric_success");
+    private void showAuthenticationScreen() {
+        KeyguardManager keyguardManager = ContextCompat
+                .getSystemService(cordova.getActivity(), KeyguardManager.class);
+        if (keyguardManager == null
+                || android.os.Build.VERSION.SDK_INT < android.os.Build.VERSION_CODES.LOLLIPOP) {
+            return;
+        }
+        if (keyguardManager.isKeyguardSecure()) {
+            Intent intent = keyguardManager.createConfirmDeviceCredentialIntent(null, null);
+            cordova.setActivityResultCallback(this);
+            cordova.getActivity().startActivityForResult(intent, REQUEST_CODE_CONFIRM_DEVICE_CREDENTIALS);
+        } else {
+            // Show a message that the user hasn't set up a lock screen.
+            sendError(
+                    new Error(
+                            PluginError.BIOMETRIC_SCREEN_GUARD_UNSECURED.getValue(),
+                            "Go to 'Settings -> Security -> Screenlock' to set up a lock screen")
+            );
+        }
     }
 
-    @Override
-    public void onAuthenticationHelp(int helpCode, CharSequence helpString) {
-        Log.e(TAG, "onAuthenticationHelp: " + helpCode + " | " + helpString);
-        sendError(helpCode, helpString != null ? helpString.toString() : "Error has no description.");
-    }
 
     @Override
-    public void onAuthenticationError(int errorCode, CharSequence errString) {
-        switch (errorCode)
-        {
-            case 7:
-            case 9:
-                if(!mDisableBackup) {
-                    showAuthenticationScreen();
-                }else{
-                    sendError(errorCode == 7 ? PluginError.BIOMETRIC_LOCKED_OUT.getValue() : PluginError.BIOMETRIC_LOCKED_OUT_PERMANENT.getValue(), errString.toString());
-                }
-                break;
-
-            default:
-                sendError(errorCode, (errString != null ? errString.toString() : "Error has no description."));
-                break;
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode == REQUEST_CODE_CONFIRM_DEVICE_CREDENTIALS) {
+            if (resultCode == Activity.RESULT_OK) {
+                sendSuccess();
```

Comment:
@exxbrain do you want to look at the failed checks..I don't mind? looks like line 277 of Fingerprint.java is missing parameter.

Reply:
Yes. Give me a second.

Reply:
Done

Reply:
Also fixed some code climate errors

Reply:
Is it possible to try again?

Reply:
Just create a pull request on my branch like before and I'll merge it in thanks.

Reply:
Done.

Reply:
I've just haven't got what does this mean:
```
Run actions/labeler@v2
2 with:
3 repo-token: ***
4 configuration-path: .github/labeler.yml
5 ##[error]HttpError: Resource not accessible by integration
6 ##[error]Resource not accessible by integration
7 ##[error]Node run failed with exit code 1
```

Reply:
lol I have no idea but its solved some how. Now ios is not passing build check ..but I don't think its related to code but pr config I'll have to check with @NiklasMerz unless you know what the issue is

Reply:
Should I fix 
```
cordova-paramedic: looking for a config here: /private/tmp/paramedic/conf/pr/ios-10.0.config.json
1386/private/tmp/paramedic/lib/utils/utilities.js:213
1387    throw new Error('Can\'t find the specified config.');
1388    ^
1389
1390Error: Can't find the specified config.
1391    at Object.getConfigPath (/private/tmp/paramedic/lib/utils/utilities.js:213:11)
1392    at Object.<anonymous> (/private/tmp/paramedic/main.js:74:34)
```
?

Reply:
I'm not sure.. how are you be able to fix this? @NiklasMerz is this something that needs to be looked at in Travis?